File: apple/Packages/JustAVPNCore/Sources/JustAVPNCore/Models/ConnectionState.swift
import Foundation

public enum ConnectionState: Equatable {
    case disconnected
    case connecting
    case connected(since: Date)
    case disconnecting
    case paused(resumeAt: Date)
    case error(String)

    public var isActive: Bool {
        switch self {
        case .connected, .connecting: return true
        default: return false
        }
    }

    public var statusText: String {
        switch self {
        case .disconnected: return "Disconnected"
        case .connecting: return "Connecting..."
        case .connected: return "Connected"
        case .disconnecting: return "Disconnecting..."
        case .paused(let date):
            let remaining = max(0, date.timeIntervalSinceNow)
            let minutes = Int(remaining) / 60
            let seconds = Int(remaining) % 60
            return "Paused (\(minutes):\(String(format: "%02d", seconds)))"
        case .error(let msg): return "Error: \(msg)"
        }
    }
}

File: android/app/src/main/kotlin/com/justavpn/data/ConnectionState.kt
package com.justavpn.data

import java.time.Instant

sealed class ConnectionState {
    data object Disconnected : ConnectionState()
    data object Connecting : ConnectionState()
    data class Connected(val since: Instant = Instant.now()) : ConnectionState()
    data object Disconnecting : ConnectionState()
    data class Paused(val resumeAt: Instant) : ConnectionState()
    data class Error(val message: String) : ConnectionState()

    val isActive: Boolean
        get() = this is Connected || this is Connecting

    val statusText: String
        get() = when (this) {
            is Disconnected -> "Disconnected"
            is Connecting -> "Connecting..."
            is Connected -> "Connected"
            is Disconnecting -> "Disconnecting..."
            is Paused -> "Paused"
            is Error -> "Error: $message"
        }
}
